data/calculated.py

```diff
@@ -127,7 +127,18 @@ def __getitem__(self, key: str) -> xr.DataArray:
         data_array = self._parser.parse(
             self._expression, self._parent, key, self._dims)
 
-        return xr.DataArray(data_array, attrs=self.attrs)
+        if hasattr(data_array, 'dims'): 
+            dims = data_array.dims
+        else: 
+            dims = self._dims
+
+        key = [k for k in key if type(k) is slice] #remove any non-slice elements 
```

Comment:
String comprehensions can get a bit hairy for understanding so just use `filter` so it's self-documenting what the general operation is doing.
```suggestion
        key = [filter(lambda k: type(k) is slice, key)]
```